### ISSUE TITLE
ORDER-EMBEDDINGS OF IMAGES AND LANGUAGE

### ISSUE THREAD

**Issue by hello-im-yj** [One-Day-One-Paper/One-Day-One-Paper#114]
# Abstract

- Hypernymy prediction(상위어 예측), textual entailment(텍스트 함의) and image-caption retrieval 등에 활용  
  ![image](https://user-images.githubusercontent.com/63237947/159168362-48bcf0f4-c94f-4cf7-8f9b-a57e0bd4da39.png)

    
    *visual-semantic hierarchy*
    
- In this paper : advocate for **explicitly modeling the partial order structure of this hierarchy.**
- Towards this goal, we 1) introduce a general method for learning ordered representations, and 2) show how it can be applied to a variety of tasks involving images and language.

# 1 INTRODUCTION

Our goal in this work is to learn representations that respect this **partial order structure.**

→ by learning a mapping which is not distance-preserving but ***order*-preserving** between the visual-semantic hierarchy and a partial order over the embedding space. We call embeddings learned in this way ***order-embeddings*.**

# 2 LEARNING ORDER-EMBEDDINGS

**partial order completion*.

- given a set of positive examples $P = {(u, v)}$ : partially ordered set $(X, ≼_X )$ 에서 가져온 ordered pairs & a set of negative examples $N$ which we know to be unordered.
- Our goal : to predict whether an unseen pair $(u′, v′)$ is ordered.

We tackle this problem by learning a mapping from X into a partially ordered embedding space (Y, ≼Y ).

*1) Order-Embedding:* Denote the mapping by *f* : (*X*,≼*X*) → (*Y*,≼*Y*). For all *u*,v ∈ *X* and the corresponding mappings *s*,*t* ∈ *Y* where *s*= *f*(*u*) and *t*= *f*(v),

$$
u≼v \text{ if and only if \ }∀i=1,...,N, s_i ≥t_i
$$

we define the penalty for an ordered pair(x,y) of points in $R^N_+$ as

$$
E(x, y) = || max(0, y − x)||^2
$$

Crucially, E(x,y) = 0 ⇐⇒ x ≼ y according to the reversed product order; if the order is not satisfied, E(x, y) is positive.

This measures the degree to which pair (x,y) violates the coordinate-wise order. It is used in the loss function during training.

*2) Loss Function:*

Recall that P and N are our positive and negative examples, respectively. Then, to learn an approximate order-embedding f, we could use a max-margin loss which encourages positive examples to have zero penalty, and negative examples to have penalty greater than a margin:

![image](https://user-images.githubusercontent.com/63237947/159168347-ba3951e9-d91c-49ec-9fa3-75aafe4fccff.png)
# 3 Experiment

👍 : On two of these tasks, hypernym prediction and caption-image retrieval, our methods outperform all previous work.

👎 : TEXTUAL ENTAILMENT / NATURAL LANGUAGE INFERENCE → our method is almost certainly worse than the state-of-the-art method of Rockta ̈schel et al. (2015), which uses a word-by-word attention mechanism, it is also much simpler.
